core/src/main/java/com/theagilemonkeys/ellmental/core/schema/Embedding.java

```diff
@@ -1,14 +1,22 @@
 package com.theagilemonkeys.ellmental.core.schema;
 
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
 /**
  * Embeddings represent a point in the embeddings space, representing the semantics of a given text.
  */
-public record Embedding(
-        UUID id,
-        List<Double> vector,
-        Map<String, String> metadata
-) {}
+@Data
+@RequiredArgsConstructor
```

Comment:
Super tiny detail but @Data includes @RequireArgsConstructor -> https://projectlombok.org/features/Data

Completely optional by the way
```suggestion

```